rust2vec/src/storage.rs

```diff
@@ -92,9 +92,11 @@ impl Storage {
         W: Write + Seek,
     {
         // n_rows: 8 bytes, n_cols: 4 bytes, type_id: 4, matrix
-        let chunk_len = 16 + (data.rows() * data.cols() * mem::size_of::<f32>());
 
         write.write_u32::<LittleEndian>(ChunkIdentifier::NdArray as u32)?;
+        let n_padding = padding::<f32>(write.seek(SeekFrom::Current(0))? + 20);
```

Comment:
Why `+20` here? First multiple of 4 above 16?

Reply:
Nope, we need the padding for the embedding matrix, but when we have just written the chunk identifier, we are not in the right position yet for computing the padding. So, we need to add the counts for everything from our current position to the start of the matrix (it should actually be 24). But I just realized that this is not necessary, since everything between the chunk length and the matrix has sizes of multiples of 4, so the amount of padding stays the same.

I'll update the PR to simplify this.